internal/oci/container_test.go

```diff
@@ -238,4 +238,38 @@ var _ = t.Describe("Container", func() {
 		// Then
 		Expect(err).To(BeNil())
 	})
+	It("should be false if pid unintialized", func() {
+		// Given
+		state := &oci.ContainerState{}
+		state.Pid = 0
+		sut.SetState(state)
+		// When
+		err := sut.IsRunning()
+
+		// Then
+		Expect(err).To(Equal(false))
+	})
+	It("should succeed if pid is running", func() {
+		// Given
+		state := &oci.ContainerState{}
+		state.Pid = 1
+		sut.SetState(state)
+		// When
+		err := sut.IsRunning()
+
+		// Then
+		Expect(err).To(Equal(true))
+	})
+	It("should be false if pid is not running", func() {
+		// Given
+		state := &oci.ContainerState{}
+		// the highest allowed pid + 1
+		state.Pid = 4194305
```

Comment:
Ideally, this should be the value from `/proc/sys/kernel/pid_max` + 1 as this is configurable.

From the other side, it seems that `4194304` (`0x400000`) is indeed the maximum value possible, so might be OK to leave as is.

```console
# echo 10000 > /proc/sys/kernel/pid_max
# cat /proc/sys/kernel/pid_max
10000
# echo 0x400000 > /proc/sys/kernel/pid_max
# cat /proc/sys/kernel/pid_max
4194304
# echo 0x400001 > /proc/sys/kernel/pid_max
bash: echo: write error: Invalid argument
```

Reply:
I don't personally think the unit tests need to be this precise

Reply:
My concern was if there are kernels out there that can have pid_max which is greater than the one you use in the test, the test would fail.

It looks like there are no such kernels (at least for now), so that is why I said it's OK to leave this as is.